nanoCAT/recipes/coordination_number.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+from itertools import combinations
+from typing import Dict, Tuple
+from scm.plams import Molecule
+from FOX import group_by_values
+from nanoCAT.bde.guess_core_dist import guess_core_core_dist
+
+mol = Molecule('perovskite.xyz')
+d_outer = 5.1 
+
+xyz = np.asarray(mol)
+
+# Construct a dictionary with atomic symbols as keys and arrays of indices as values
+elements = [at.symbol for at in mol.atoms]
+symbol_enumerator = enumerate(elements)
+idx_dict = group_by_values(symbol_enumerator)
+for k, v in idx_dict.items():
+    idx_dict[k] = np.fromiter(v, dtype=int, count=len(v))
+
+def idx_pairs(idx_dict: Dict[str, np.ndarray]) -> Tuple[np.ndarray, np.ndarray]:    
```

Comment:
Yup, I feel this is a good case for a function!

I'd recommend contuining partitioning the script into functions and then basically have one "main" functions which distributes and manages the workflow.